scripts/install/lib/common.sh

```
@@ -0,0 +1,536 @@
+# Check if a command exists
+# Usage: if command_exists "git"; then echo "git found"; fi
+command_exists() {
+    local cmd="${1:-}"
+    command -v "$cmd" >/dev/null 2>&1
+}
+
+# Require a command to exist, die if not found
+# Usage: require_command "git" "Git is required for version control"
+require_command() {
+    local cmd="${1:-}"
+    local message="${2:-Command '$cmd' is required but not installed}"
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)